From 52e294c5737212e56bf440e79f39bcae535f0e05 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Mon, 4 May 2026 23:56:18 +0800
Subject: [PATCH 1/2] AOSCOS: patching config to apply best practices

- Source /etc/ssh/ssh_config.d/* and /etc/ssh/sshd_config.d/* with the
  highest precedence.
- Enable PAM authentication.
- Disable PrintMotd as /etc/motd printing is already handled by PAM.
- Hash known_hosts entries.

Signed-off-by: Student Main <studentmain@aosc.io>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 ssh_config  |  6 +++++-
 sshd_config | 10 ++++++++--
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/ssh_config b/ssh_config
index 238a0c5e3..ab688c9df 100644
--- a/ssh_config
+++ b/ssh_config
@@ -17,7 +17,10 @@
 # list of available options, their meanings and defaults, please see the
 # ssh_config(5) man page.
 
-# Host *
+# Extra configurations take precedence.
+Include /etc/ssh/ssh_config.d/*
+
+Host *
 #   ForwardAgent no
 #   ForwardX11 no
 #   PasswordAuthentication yes
@@ -43,3 +46,4 @@
 #   ProxyCommand ssh -q -W %h:%p gateway.example.com
 #   RekeyLimit 1G 1h
 #   UserKnownHostsFile ~/.ssh/known_hosts.d/%k
+    HashKnownHosts yes
diff --git a/sshd_config b/sshd_config
index 0f4a3a724..2a54f67b1 100644
--- a/sshd_config
+++ b/sshd_config
@@ -10,6 +10,9 @@
 # possible, but leave them commented.  Uncommented options override the
 # default value.
 
+# Extra configurations take precedence.
+Include /etc/ssh/sshd_config.d/*
+
 #Port 22
 #AddressFamily any
 #ListenAddress 0.0.0.0
@@ -81,7 +84,7 @@ AuthorizedKeysFile	.ssh/authorized_keys
 # If you just want the PAM account and session checks to run without
 # PAM authentication, then enable this but set PasswordAuthentication
 # and KbdInteractiveAuthentication to 'no'.
-#UsePAM no
+UsePAM yes
 
 #AllowAgentForwarding yes
 #AllowTcpForwarding yes
@@ -90,7 +93,10 @@ AuthorizedKeysFile	.ssh/authorized_keys
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PermitTTY yes
-#PrintMotd yes
+
+# Prevent /etc/motd from being printed twice
+PrintMotd no
+
 #PrintLastLog yes
 #TCPKeepAlive yes
 #PermitUserEnvironment no
-- 
2.55.0

